File: BookShop.WebAPI/Models/Book.cs
using System;

namespace BookShop.WebAPI.Models
{
    public class Book : BaseEntity
    {
        public DateTime DateAdded { get; set; }
        public DateTime DateRelease { get; set; }
        public bool Opportunity { get; set; }
        public decimal Price { get; set; }
        public string Title { get; set; }

        public virtual Author Author { get; set; }
        public virtual Category Category { get; set; }
        public virtual PublishingHouse PublishingHouse { get; set; }
    }
}

File: BookShop.Client/src/app/models/Book.ts
import { Author } from './Author'
import { Carrier } from './Carrier'
import { PublishingHouse } from './PublishingHouse'

export class Book {
    DateAdded: Date;
    DateRelease: Date;
    Opportunity: boolean;
    Price: number;
    Title:string;

    Author: Author;
    Category: Carrier;
    PublishingHouse: PublishingHouse;
}